File: data/code_pairs/fortran/1512.f90
program test_transfer
implicit none

    ! Variables
    integer :: i
    logical :: b

    ! Test transferring int to bool
    i = 0
    b = transfer(i, .true.)
    print *, 'Transfer int 0 to bool:', b

    ! Test transferring nonzero int to bool
    i = -1
    b = transfer(i, .true.)
    print *, 'Transfer int -1 to bool:', b

end program test_transfer

File: data/code_pairs/cpp/1512.cpp
#include <iostream>
#include <cstring>

int main() {
    // Variables
    int i;
    bool b;

    // Test transferring int to bool
    i = 0;
    std::memcpy(&b, &i, sizeof(b));
    std::cout << "Transfer int 0 to bool: " << std::boolalpha << b << std::endl;

    // Test transferring nonzero int to bool
    i = -1;
    std::memcpy(&b, &i, sizeof(b));
    std::cout << "Transfer int -1 to bool: " << std::boolalpha << b << std::endl;

    return 0;
}